harness/determined/cli/rbac.py

```diff
@@ -0,0 +1,372 @@
+            role_id_to_permissions[r.roleId] = set(r.permissions)
+
+    scope_id_to_permissions: Dict[int, Set[bindings.v1Permission]] = {}
+    for a in resp.assignments:
+        if a.roleId is None:
+            raise api.errors.BadResponseException("expected roleId to be provided")
+
+        if a.isGlobal:
+            if 0 not in scope_id_to_permissions:
+                scope_id_to_permissions[0] = set()
+            scope_id_to_permissions[0].update(role_id_to_permissions[a.roleId])
+
+        if a.scopeWorkspaceIds is None:
+            a.scopeWorkspaceIds = []
+        for wid in a.scopeWorkspaceIds:
+            if wid not in scope_id_to_permissions:
+                scope_id_to_permissions[wid] = set()
+            scope_id_to_permissions[wid].update(role_id_to_permissions[a.roleId])
+
+    if len(scope_id_to_permissions) == 0:
+        print("no permissions assigned")
+        return
+    for wid, perms in scope_id_to_permissions.items():
+        if wid == 0:
+            print("global permissions assigned")
+        else:
+            workspace_name = bindings.get_GetWorkspace(session, id=wid).workspace.name
+            print(f"permissions assigned over workspace '{workspace_name}' with ID '{wid}'")
+        for p in perms:
+            print(f"\tpermission '{p.name}' with type {p.id}")
+        print()
+
+
+@authentication.required
+@require_feature_flag("rbacEnabled", rbac_flag_disabled_message)
+def list_roles(args: Namespace) -> None:
+    req = bindings.v1SearchRolesAssignableToScopeRequest(
+        limit=args.limit,
+        offset=args.offset,
+        workspaceId=1 if args.exclude_global_roles else None,
+    )
+    resp = bindings.post_SearchRolesAssignableToScope(setup_session(args), body=req)
+    if args.json:
+        print(json.dumps(resp.to_json(), indent=2))
+        return
+
+    if resp.roles is None or len(resp.roles) == 0:
+        print("no roles found")
+        return
+    for r in resp.roles:
+        print(f"role '{r.name}' with ID {r.roleId}")
+        if r.permissions is None:
+            continue
+        for p in r.permissions:
+            print(f"\twith{' global' if p.isGlobal else ''} permission '{p.name}' with type {p.id}")
+
+
+@authentication.required
+@require_feature_flag("rbacEnabled", rbac_flag_disabled_message)
+def list_users_roles(args: Namespace) -> None:
+    session = setup_session(args)
+    user_id = user_groups.usernames_to_user_ids(session, [args.username])[0]
+    resp = bindings.get_GetRolesAssignedToUser(session, userId=user_id)
+    if args.json:
+        print(json.dumps(resp.to_json(), indent=2))
+        return
+
+    if resp.roles is None or len(resp.roles) == 0:
+        print("user has no role assignments")
+        return
+    for r in resp.roles:
+        print(f"user is assigned to role '{r.name}' with ID {r.roleId}")
+        if r.permissions is None:
+            continue
+        for p in r.permissions:
```

Comment:
I think we should omit permissions here - it makes sense to me just to return the roles associated with a user and leave the permissions as a permission lookup.